module/zfs/zfs_ioctl.c

```diff
@@ -219,6 +232,33 @@ typedef int zfs_ioc_legacy_func_t(zfs_cmd_t *);
 typedef int zfs_ioc_func_t(const char *, nvlist_t *, nvlist_t *);
 typedef int zfs_secpolicy_func_t(zfs_cmd_t *, nvlist_t *, cred_t *);
 
+/*
+ * IOC Keys are used to document and validate user->kernel interface inputs.
+ * See zfs_keys_recv_new for an example declaration. Any key name that is not
+ * listed will be rejected as input.
+ *
+ * The keyname 'optional' is always allowed, and must be an nvlist if present.
+ * Arguments which older kernels can safely ignore can be placed under the
+ * "optional" key.
+ *
+ * In general, if you add new keys to an existing ioc for new functionality,
+ * you should also add an entry into zfs_sysfs.c zfs_features[] list.
```

Comment:
...and update the `libzfs_input_check.c` test utility.

Reply:
fixed